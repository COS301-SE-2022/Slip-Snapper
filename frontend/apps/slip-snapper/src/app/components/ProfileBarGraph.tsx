import { Bar } from "react-chartjs-2";

type Props = {
    graphData: any;
}
function ProfileBarGraph({ graphData }: Props) {
    const graphSettings = {
        responsive: true,
        barThickness: 43,
        borderWidth: 1,
        borderRadius: 10,
        color: 'white',


        plugins: {
            legend: {
                position: 'bottom' as const,
            },
        },

        scales: {
            yAxes: {
                ticks: {
                    beginAtZero: true,
                    color: 'white',
                    fontSize: 12,
                }
            },
            xAxes: {
                ticks: {
                    beginAtZero: true,
                    color: 'white',
                    fontSize: 12,
                }
            },
        },
    };

    const graphStats = {
   
        datasets: [
            {
                label: 'Expenditure (R)',
                data: graphData,
                backgroundColor: "rgb(39, 165, 146, 0.8)",
                width: 8,
                hoverBackgroundColor: "rgb(47, 198, 176, 0.8)",
                
            }
        ],
    };
    return (
        <Bar className="graph" options={graphSettings} data={graphStats} />
    );
}

export default ProfileBarGraph;