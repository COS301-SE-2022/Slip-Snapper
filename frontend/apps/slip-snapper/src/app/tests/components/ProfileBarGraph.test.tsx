import renderer from 'react-test-renderer';
import { Bar } from "react-chartjs-2";
import ProfileBarGraph from "../../components/ProfileBarGraph";

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('<ProfileBarGraph>', () => {
  it('should render component with props', () => {
    expect(renderTree(<ProfileBarGraph  
      graphData={{current: 410.49, previous: 492.99}} 
    />).toJSON()).toMatchSnapshot();
  });
});