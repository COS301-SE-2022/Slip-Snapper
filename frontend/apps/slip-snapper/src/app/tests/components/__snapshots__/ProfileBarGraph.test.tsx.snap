// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileBarGraph> should render component with props 1`] = `
<canvas
  className="graph"
  height={250}
  role="img"
  width={300}
/>
`;
